spec/httparty/request_spec.rb

```diff
@@ -784,6 +784,260 @@
     end
   end
 
+  describe "a request that 307 redirects" do
```

Comment:
It would cause a lot of extra code in that tests file, but no real harm in that really... I'll add testing for the others.